GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2568,7 +2568,8 @@ private void UpdateSubmodulesList()
 
                     var smi = new SubmoduleInfo { Text = name, Path = path };
                     result.OurSubmodules.Add(smi);
-                    GetSubmoduleStatusAsync(smi, cancelToken);
+                    ThreadHelper.JoinableTaskFactory.RunAsync(() =>
```

Comment:
❗️ This code is already executing inside a `RunAsync` (see about 15 lines above). Nesting is unnecessary and inefficient; use the following instead:

```csharp
GetSubmoduleStatusAsync(smi, cancelToken).FileAndForget();
```

📝 Additional note about this:

1. Nesting is not inherently a problem, but it should be avoided within a single method, where the nesting is obvious and unavoidable. For cross-method nesting situations, it's better to be explicit since it helps with clarity and maintainability.
2. I only checked this location. Other lines in this PR may need similar consideration.